### PR TITLE
Address new custom build logic issues

### PR DESCRIPTION
**Pull request by m4gr3d**
Resolves several issues introduced during the addition of `AAB` support to the Android export workflow:

- Commit e06301c resolves issue https://github.com/godotengine/godot/issues/42793
- Commit e79192e resolves issue https://github.com/godotengine/godot/issues/42795
- Commit 82490f7 resolves issue https://github.com/godotengine/godot/issues/42792

### CODE REVIEW

File: platform/android/export/export.cpp [hunk 1]
```diff
@@ -2225,25 +2227,34 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
...
 		if (retval) {
-			EditorNode::add_io_error("'jarsigner' verification of APK failed. Make sure to use a jarsigner from OpenJDK 8.");
+			EditorNode::add_io_error("'jarsigner' verification of " + export_label + " failed. Make sure to use a jarsigner from OpenJDK 8.");
 			return ERR_CANT_CREATE;
 		}
 		return OK;
 	}
 
+	void _clear_assets_directory() {
+		DirAccess *da_res = DirAccess::create(DirAccess::ACCESS_RESOURCES);
```
Comment (akien-mga) [inline platform/android/export/export.cpp:2262]: I think you need to `memdelete` the DirAccess once you're done with it, or use DirAccessRef which should free it when going out of scope.

File: platform/android/export/export.cpp [hunk 2]
```diff
@@ -2323,13 +2334,8 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 			_write_tmp_manifest(p_preset, p_give_internet, p_debug);
 
 			//stores all the project files inside the Gradle project directory. Also includes all ABIs
+          _clear_assets_directory();
```
Comment (akien-mga) [inline platform/android/export/export.cpp]: Indentation.

File: platform/android/export/export.cpp [hunk 3]
```diff
@@ -2334,7 +2334,7 @@ class EditorExportPlatformAndroid : public EditorExportPlatform {
 			_write_tmp_manifest(p_preset, p_give_internet, p_debug);
 
 			//stores all the project files inside the Gradle project directory. Also includes all ABIs
-          _clear_assets_directory();
+			_clear_assets_directory();
```
Comment (akien-mga) [inline platform/android/export/export.cpp:2401]: This should ideally be amended in the first commit to avoid unnecessary diffs

Comment (m4gr3d) [reply platform/android/export/export.cpp:2401]: Done!

File: platform/android/export/export.cpp (hunk 1 above)
Comment (m4gr3d) [reply platform/android/export/export.cpp]: Done!

File: platform/android/export/export.cpp (hunk 2 above)
Comment (m4gr3d) [reply platform/android/export/export.cpp]: Done!

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**